torch/autograd/_functions/initializers.py

```diff
@@ -0,0 +1,20 @@
+from ..function import InplaceFunction
+
+
+class Zero(InplaceFunction):
+
+    @staticmethod
+    def forward(ctx, i, inplace=False):
+        if inplace:
+            ctx.mark_dirty(i)
+            result = i.zero_()
+        else:
+            result = i.new(i.size()).zero_()
+
+        ctx.save_for_backward(result)
+        return result
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        result, = ctx.saved_variables
+        return result.new(1).expand_as_(result).zero_()
```

Comment:
`return result.new([0]).expand_as_(result)`

Much more efficient

Reply:
expand_as_ + zero_ in that order will zero the whole expanded tensor (kind of a redundant operation, but that's what it does)

Reply:
ah yes - gotcha. :man_facepalming: fixed now.